src/object/cli_obj.c

```diff
@@ -6506,14 +6507,26 @@ dc_obj_query_key(tse_task_t *api_task)
 
 non_leader:
 		/* Then Try non-leader shards */
+		D_ASSERT(obj_is_ec(obj));
 		start_shard = i * obj_get_grp_size(obj);
```

Comment:
here the start shard seems possibly incorrect if parity shard rotate enabled? (as it possibly is not the first data shard)

Reply:
ok, seems fine here.